lib/ansible/modules/cloud/amazon/api_gateway.py

```diff
@@ -0,0 +1,161 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '0.1'}
+
+DOCUMENTATION = '''
+---
+module: api-gateway
+short_description: Manage AWS API Gateway APIs
+description:
+     - Allows for the management of API Gatway APIs.
+version_added: '2.2'
+requirements: [ boto3 ]
+options:
+  api_id:
+    description:
+      - The ID of the API you want to manage
+    required: true
+  state:
+    description:
+      - NOT IMPLEMENTED Create or delete Lambda function
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  api_file:
+    description:
+      - JSON or YAML file containing swagger definitions for API
+    required: true
+  stage:
+    description:
+      - stage API should be deployed to
+    required: false
+author:
+    - 'Michael De La Rue (@mikedlr)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Update API resources for development
+tasks:
+- name: update API
+  lambda:
+    api_id: 'abc123321cba'
+    state: present
+    api_file: my_api.yml
+
+# update definitions and deploy API to production
+tasks:
+- name: deploy API
+  lambda:
+    api_id: 'abc123321cba'
+    state: present
+    api_file: my_api.yml
+    stage: production
+    deploy_desc: Make auth fix available.
+'''
+
+RETURN = '''
+output:
+  description: the data returned by put_restapi in boto3
+  returned: success
+  type: dict
+  sample:
+    'data':
+      {
+          "id": "abc123321cba",
+          "name": "MY REST API",
+          "createdDate": 1484233401
+      }
+'''
+
+# Import from Python standard library
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        api_id=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        api_file=dict(type='str', default=None, aliases=['src']),
+        stage=dict(type='str', default=None),
+        deploy_desc=dict(type='str', default=None),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False,  ) #TODO !!!`
+
+    api_id = module.params.get('api_id')
+    state = module.params.get('state').lower()
+    api_file = module.params.get('api_file')
```

Comment:
For filepaths (such as `api_file`) make sure to use `expanduser` so that paths like `~/code/apis/my_fancy_api.yml` work as users expect.